h2o-py/h2o/model/model_base.py

```diff
@@ -319,7 +319,7 @@ def summary(self):
         """Print a detailed summary of the model."""
         model = self._model_json["output"]
         if model["model_summary"]:
-            model["model_summary"].show()  # H2OTwoDimTable object
+            return model["model_summary"]
 
```

Comment:
Theoretically there is no summary if there is no model. But we can return an exception here that is a bit more informative than a comment that I posted on a PR.